### ISSUE TITLE
[Feature Request] - Automation of the download cycle

### ISSUE THREAD

**Issue by SpudmanWP** [Tribler/tribler#8357]
I'd like to set up my Tribler instance to be as hands-off as possible.
It's currently in it's own Linux VM but will be in a Docker container when that gets released for 8.x

To that end, I'd like to propose some features to support automation:
1. Given that Tribler already supports subfolders in the Watch folder, let those folders represent a "Category".
2. Have "Category" appear in a UI column for the download.
3. When a file completes, move the Watch folder torrent file to a "completed" folder that mimics the Watch folder, including Category subfolders.
4. When a file completes, move the completed download folder to another location according to a rules-based filter (based on Category, Name, etc).  The file should still be able to be shared after moving.

**Comment by qstokkink** [comment]
If you want this level of complexity, it's probably a better to make your own script.

Tribler supports a REST API that allows you directly read and modify your downloads. You can play around with this by browsing to `http://127.0.0.1:12345/docs` (where `127.0.0.1` and `12345` are the IP and port Tribler is running on).

For example, moving a download with infohash `abcdefabcdefabcdef00` can be done with an HTTP PATCH request to `http://127.0.0.1:12345/api/downloads/abcdefabcdefabcdef00` and the request body:

```json
{
    "state": "move_storage",
    "dest_dir": "/some/directory"
}
```

**Comment by armchairancap** [comment]
> If you want this level of complexity, it's probably a better to make your own script.
> 

Makes sense. Such automation belongs to workflow/automation tools and seems out of scope for Tribler.

I created this simple example to get people started.
https://github.com/armchairancap/magnet-link-downloader-tribler-firefox

Notes:

- It doesn't automate post-download steps, as that was out of scope for me and is highly workflow-dependent
- What this does is initiate Tribler API  (download) requests from the browser,and it shows how the API could be used into implement such workflows and rules in a Firefox browser extension. (That would be out of scope for this extension as well; personally I'd prefer to implement it in another service that would manage them and run workflows).

![Image](https://github.com/user-attachments/assets/30da7f95-809a-423f-b8b2-cc412daa4b11)

- If someone wonders what's the problem with copying links manually and pasting them into Tribler, that was a tool for [a chat integration](https://github.com/armchairancap/xx-tribler-haven) PoC.

![Image](https://github.com/user-attachments/assets/c94ca8a6-18ea-4168-8a25-d7eb90f81bed)

- Another use case is stand-alone, for headless Tribler.

Not to sound completely negative, I think there's some room for (scoped-down) improvement, i.e. if one could add tags to downloads, filtering by some categories could be helpful to API clients that wish to automate workflows. Presently, without tags, I'd probably go with a small Web/DB combo to manage tags and workflows. Not very complicated, but more complicated than a 200 LoC Python script running periodically from `cron`. 

Another idea I've tested was an upload/download bot for the chat I mentioned above. I haven't published that code, but the way it worked is exactly as one may expect - the bot shares and posts magnet links to private chats, and can also download links posted by others (which is OK if in a community of friends or relatives) using Tribler.

**Comment by qstokkink** [comment]
I think our underlying database still supports tags for torrents and I believe we could expose that again. So, I think it's feasible to support tags. I'll schedule this for 8.2.

**Comment by qstokkink** [comment]
I just dug through Tribler's historical code. If I understand correctly, the following is the case:

 - Originally, the database entry `tags` was used for a comma-separated list of tags, as a string.
 - This `tags` string was later repurposed to capture only a single `category` (no longer comma-separated). This is what database retrieval is based on now:

    https://github.com/Tribler/tribler/blob/ba1aef3ee326fec54e8817abbde92dff26654005/src/tribler/core/database/store.py#L606

    As well as the GUI:

    https://github.com/Tribler/tribler/blob/ce7a868ea3be12545712d3a3f481c6b41221331f/src/tribler/ui/src/lib/utils.ts#L36

Based on that, my plan is the following:

 - Interpret the `tags` column in the database table as a comma-separated list again.
 - Add REST API calls to add and remove tags.
 - Only forward a single "category" tag (from [the list of known category tags](https://github.com/Tribler/tribler/blob/ce7a868ea3be12545712d3a3f481c6b41221331f/src/tribler/ui/src/lib/utils.ts#L38-L51)), if it exists, to the GUI. So, the GUI can remain unchanged.
